src/processor/reconnection-graph.service.ts

```diff
@@ -171,36 +171,35 @@ export class ReconnectionGraphService {
   }
 
   async getUserGraphFromProvider(dsnpUserId: MessageSourceId | string, providerId: ProviderId | string): Promise<any> {
-    const headers = {
-      Authorization: 'Bearer <access_token>', // Replace with your actual access token if required
-    };
-    const baseUrl = this.configService.providerBaseUrl(providerId);
+    const providerAPI = this.getProviderAPI(providerId);
 
     const params = {
       pageNumber: 1,
       pageSize: 10, // This likely should be increased for production values
     };
 
-    const providerAPI: AxiosInstance = axios.create({
-      baseURL: baseUrl.toString(),
-      headers,
-    });
-
     const allConnections: ProviderGraph[] = [];
     const keyPairs: GraphKeyPair[] = [];
-    try {
-      let hasNextPage = true;
-      while (hasNextPage) {
-        // eslint-disable-next-line no-await-in-loop
+
+    let hasNextPage = true;
+    let webhookFailures: number = 0;
+    const webhookFailureThreshold: number = this.configService.getWebhookFailureThreshold();
+    const webhookRetryIntervalMilliseconds: number = this.configService.getWebhookRetryIntervalSeconds() * 1000;
+
+    while (hasNextPage) {
+      // eslint-disable-next-line no-await-in-loop
+      try {
         const response = await providerAPI.get(`/api/v1.0.0/connections/${dsnpUserId.toString()}`, { params });
-        if (response.status !== 200) {
-          throw new Error(`Bad status ${response.status} (${response.statusText} from Provider web hook.)`);
-        }
+
+        // Reset webhook failures to 0 on a success. We don't go into waiting for recovery unless
+        // a sequential number failures occur equaling webhookFailureThreshold.
+        webhookFailures = 0;
+
         if (!response.data || !response.data.connections) {
           throw new Error(`No connections found for ${dsnpUserId.toString()}`);
         }
```

Comment:
Not sure what we want to do here... I guess if these are missing it's a malformed response. Perhaps we should track these somehow? If the provider's endpoint is misconfigured, we don't want to keep trying & failing to process jobs.

I'll add it as an error scenario we want to handle. For now, maybe just tweak the error message to be more illuminating, like `Invalid response from provider` or something?

Reply:
☝🏻 Added to #64